internal/endpoints/endpointsFetcher_test.go

```diff
@@ -0,0 +1,12 @@
+package endpoints
```

Comment:
i know the task was "figure out testing", but since we're doing a PR of this, flesh this out to be a complete set of tests for the initial set of code.